lib/compiler-finder.js

```diff
@@ -146,6 +146,48 @@ class CompilerFinder {
         }));
     }
 
+    copyAndFilterLibraries(allLibraries, filter) {
+        const filterLibIds = [];
```

Comment:
Does a Set() help here? I'm not sure it needs to be ordered, and saves the `if (!includes) append` bit. No worries either way.

Maybe also consider doing this in two stages: the filterLibAndVersoin, then separately afterwards `map()`-ing over that to pull out the `id` field. It's a little unconventional to have a side-effecting `map()` operation, though it looks like it works to me :)

Reply:
I didn't like the side-effect either. Assuming there won't be much to iterate over, a separate operation would be fine.
Will also try out the set.
Thanks!